_posts/2025-05-21-geospatial-techno.md

```diff
@@ -0,0 +1,88 @@
+
+**Note.** This guide outlines the installation of GeoServer 2.25.x using Java 17 and Apache Tomcat 9, followed by upgrade instructions. To ensure you have the latest release, please visit this [link](https://geoserver.org/download/) and avoid using older versions of GeoServer.
+
+## Preparing for Installation
+Before proceeding, follow the steps below:
+- Backup the existing GeoServer folder (if upgrading).  
+- Check the **Modules** tab under the **Server Status** page to see all installed extensions.
+- Uninstall previous versions of Java and Apache Tomcat.
+
+### Installing Java Development Kit (JDK)
+To download JDK 17, navigate to [adoptium.net](https://adoptium.net) and select:  
+   - **Operating System:** Windows  
+   - **Architecture:** x64  
+   - **Package Type:** JDK  
+   - **Version:** 17-LTS  
+
+Download the `.msi` file and run it as an administrator. During installation, accept default settings and complete the setup.
+
+## Installing Apache Tomcat
+To download and install Apache Tomcat software, navigate to [tomcat.apache.org](https://tomcat.apache.org) and select **Tomcat 9** from the **Download** section.
+
+Choose the **32-bit/64-bit Windows Service Installer** and run it as an administrator.
+
+During setup:  
+   - Configure the **ports** (default recommended).  
+   - Set a **secure** username and password for administration (avoiding common defaults like `admin` or `tomcat`).  
+   - The installer should auto-detect the installed JDK; if not, the user manually selects the Java installation path.   
+
+To configure JVM memory allocation, navigate to `C:\Program Files\Apache Software Foundation\Tomcat 9.0\bin` and run **Tomcat9w.exe** as an administrator.
+
+In the **Java** tab, the user sets:  
+   - **Initial Memory Pool:** 512 MB
+   - **Maximum Memory Pool:** 1024 MB
+   
+Switch to the **General** tab, and set **Startup Type** to **Automatic**, and start the Tomcat service.
+
+## Deploying GeoServer
+Download the latest **GeoServer WAR** file from [geoserver.org](https://geoserver.org).  
+
+Extract the `.war` file and copy it to `C:\Program Files\Apache Software Foundation\Tomcat 9.0\webapps`.
+  
+To start GeoServer:  
+   - Navigate `http://localhost:8080/manager`.  
+   - Login with the Tomcat credentials.  
+   - Click **Start** next to the GeoServer application.
+   
+The user accesses GeoServer at `http://localhost:8080/geoserver` and logs in using the default credentials:  
+   - **Username:** admin  
+   - **Password:** geoserver  
+
+## Upgrading GeoServer
+Stop GeoServer via the **Tomcat Manager App**, then replace the existing `geoserver` directory in `webapps` with the backup.
+
+Reinstall any **compatible extensions** for the new version, and restart GeoServer and verifies functionality.  
```

Comment:
This is incorrect the ``webapps/geoserver`` folder is the result of unpacking ``geoserver.war``. By replacing the folder you would be reverting to the previous version.

Please see the upgrade instructions: https://docs.geoserver.org/latest/en/user/installation/upgrade.html

1. backup the *data directory**, if you previously just installed you will need to copy this to an external location and use a system property or environmental variable. See https://docs.geoserver.org/latest/en/user/datadirectory/index.html

2. Adding the geoserver.war, and starting up tomcat again will "deploy" geoserver, recreating the geoserver folder and replacing what was already there. This is why it is important to take the data directory out of this location or you would lose your configuration.

Reply:
I have provided a suggestion above.